lib/puppet/type/service.rb

```diff
@@ -38,6 +38,12 @@ module Puppet
     feature :enableable, "The provider can enable and disable the service.",
       :methods => [:disable, :enable, :enabled?]
 
+    feature :delayed_startable, "The provider can set service to delayed start",
+      :methods => [:delayed_start]
+
+    feature :manual_startable, "The provider can set service to manual start",
+      :methods => [:manual_start]
```

Comment:
To preserve the windows behaviour wouldn't be necessary to add these features to the windows service provider https://github.com/puppetlabs/puppet/blob/main/lib/puppet/provider/service/windows.rb#L17?

Maybe add some tests on that provider to reflect that it can be used with `enable: delayed|manual` if there aren't any

Reply:
Any provider that implements the `delayed_start` and `manual_start` methods will automatically have the `delayed_startable` and `manual_startable` features added.